optaplanner-examples/src/main/java/org/optaplanner/examples/pas/solver/score/PatientAdmissionMoveConstraintProvider.java

```diff
@@ -0,0 +1,180 @@
+        };
+    }
+
+    public Constraint sameBedInSameNightConstraint(ConstraintFactory constraintFactory) {
+        return constraintFactory.fromUniquePair(BedDesignation.class)
+                .filter((left, right) -> left.getBed() != null
+                        && right.getBed() != null
+                        && left.getBed() == right.getBed()
+                        && left.getAdmissionPart().calculateSameNightCount(right.getAdmissionPart()) > 0)
+                .penalize("sameBedInSameNight", HardMediumSoftScore.ofHard(1000),
+                        (leftBd, rightBd) -> leftBd.getAdmissionPart().calculateSameNightCount(rightBd.getAdmissionPart()));
+    }
+
+    public Constraint femaleInMaleRoomConstraint(ConstraintFactory constraintFactory) {
+        return constraintFactory.from(BedDesignation.class)
+                .filter(bd -> bd.getPatientGender() == Gender.FEMALE
+                        && bd.getRoomGenderLimitation() == GenderLimitation.MALE_ONLY)
+                .penalize("femaleInMaleRoom", HardMediumSoftScore.ofHard(50), BedDesignation::getAdmissionPartNightCount);
+    }
+
+    public Constraint maleInFemaleRoomConstraint(ConstraintFactory constraintFactory) {
+        return constraintFactory.from(BedDesignation.class)
+                .filter(bd -> bd.getPatientGender() == Gender.MALE
+                        && bd.getRoomGenderLimitation() == GenderLimitation.FEMALE_ONLY)
+                .penalize("maleInFemaleRoom", HardMediumSoftScore.ofHard(50), BedDesignation::getAdmissionPartNightCount);
+    }
+
+    public Constraint differentGenderInSameGenderRoomInSameNightConstraint(ConstraintFactory constraintFactory) {
+        return constraintFactory.from(BedDesignation.class)
+                .filter(bd -> bd.getRoomGenderLimitation() == GenderLimitation.SAME_GENDER
+                        && bd.getBed() != null)
+                .join(BedDesignation.class,
+                        equal(BedDesignation::getRoom),
+                        lessThan(BedDesignation::getId),
+                        filtering((left, right) -> right.getRoomGenderLimitation() == GenderLimitation.SAME_GENDER
+                                && left.getAdmissionPart().calculateSameNightCount(right.getAdmissionPart()) > 0
+                                && left.getPatient().getGender() != right.getPatient().getGender()))
+                .penalize("differentGenderInSameGenderRoomInSameNight", HardMediumSoftScore.ofHard(1000),
+                        (left, right) -> left.getAdmissionPart().calculateSameNightCount(right.getAdmissionPart()));
+    }
+
+    public Constraint departmentMinimumAgeConstraint(ConstraintFactory constraintFactory) {
+        return constraintFactory.from(Department.class)
+                .filter(d -> d.getMinimumAge() != null)
+                .join(BedDesignation.class,
+                        equal(d -> d, BedDesignation::getDepartment),
```

Comment:
`d -> d` should be the same as `Function.identity()`.

Reply:
Fix 94c60080

Reply:
Fix https://github.com/kiegroup/optaplanner/pull/872/commits/94c60080ed05dd144db456a18e268b029fdd3328